src/flacarray/array.py

```diff
@@ -240,45 +240,85 @@ def _keep_view(self, key):
         view[key] = True
         return view
 
+    def _slice_nelem(self, slc, dim):
+        islc = slc.indices(dim)
+        nslc = (islc[1] - islc[0]) // islc[2]
+        if islc[1] < islc[0]:
+            nslc += 1
+        return nslc
```

Comment:
"indices" is not as helpful as it should be ...
```
import numpy as np

def _slice_nelem(slc, dim):
    islc = slc.indices(dim)
    nslc = (islc[1] - islc[0]) // islc[2]
    if islc[1] < islc[0]:
        nslc += 1
    return nslc

test = np.zeros(20)

for slc in [
        slice(0),
        slice(10, 15),
        slice(10, 5),
        slice(10, 5, -1),
]:
    print(_slice_nelem(slc, len(test)), len(test[slc]))
```

Produces
```
0 0
5 5
-4 0
6 5
```

So those last two rows are probably not what you were hoping to get.

Reply:
Ok, I simplified `__getitem__()` a bit and changed it to hand these sorts of cases.  If any dimension of the returned array has zero elements, then an empty array of the correct shape and type is returned.  I also added these types of slices to the unit test.  Maybe third time is the charm on this PR :-)